### ISSUE TITLE
something different about -oJ  json format

### ISSUE THREAD

**Issue by JohnsonYan** [robertdavidgraham/masscan#314]
Hello,
In Json format output, if the last line end with  ',' ,  when parsing json file, I got an error msg.

**Comment by joydragon** [comment]
+1

**Comment by schrodyn** [comment]
I have this issue too, rather annoying when trying to parse JSON reports. I have had to create a tool to fix each output file.

**Comment by zeepi222** [comment]
+1 Having same issue with the trailing ',' in the JSON output
https://jsonlint.com also considers the ouput as invalid JSON

**Comment by BBerastegui** [comment]
+1 here. The json output is not properly formatted.

I made this ugly hack in my python post-processing for cleaning that last trailing comma:

```"".join(raw_data.split()).rstrip(",]") + str("]")```

**Comment by mzpqnxow** [comment]
Agree that the trailing comma needs to be fixed, it’s very annoying and I
am among many who have submitted PRs to fix this because it’s extremely
annoying.

That said, I’d like to suggest that the format remain as JSON-Lines format
as it original ally was early on in development, rather than one JSON blob.
It leads to less memory pressure and general performance issues when
loading a huge scan all in one json.load() operation, like when your JSON
file is 100MB.

Basically, the original format that everyone complained about as being
invalid JSON, which was JSON-Lines. Sorry, it’s off topic, but worth
mentioning. Processing one line at a time for huge scans seems to be
surprisingly faster in my experience though I suppose it depends on your
JSON implementation and language.

This format is what masscan originally used, in case anyone doesn’t
know/recall.

**Comment by ravenium** [comment]
Fixed in 1.0.5+, per release notes.

**Comment by paalbra** [comment]
I'm confused when it comes to the versioning. Is this issue fixed in some release/commit or not? :S

On CentOS 8:
```
[root@8d6b7c21b06f ~]# dnf list --installed masscan
Installed Packages
masscan.x86_64                         1.0.5-2.el8                         @epel
[root@8d6b7c21b06f ~]# masscan --version

Masscan version 1.0.4 ( https://github.com/robertdavidgraham/masscan )
Compiled on: Jan 13 2020 14:16:23
Compiler: gcc 8.3.1 20190507 (Red Hat 8.3.1-4)
OS: Linux
CPU: unknown (64 bits)
GIT version: unknown
```

@superorc comment above mention both `Masscan version 1.0.6` and `GIT version: 1.0.5-86-ga025970`

**Comment by paalbra** [comment]
Looks like it's resolved on a Fedora machine with:
```
$ dnf list --installed masscan
Installed Packages
masscan.x86_64                        1.3.1-2.fc33                        @updates
$ masscan --version

Masscan version 1.3.1 ( https://github.com/robertdavidgraham/masscan )
Compiled on: Jan 27 2021 00:00:00
Compiler: gcc 10.2.1 20201125 (Red Hat 10.2.1-9)
OS: Linux
CPU: unknown (64 bits)
GIT version: unknown
```
